pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -667,6 +670,268 @@ func TestGetPipelineFunc_RemoteResolutionInvalidData(t *testing.T) {
+			requester:                requesterSigned,
+			resourceVerificationMode: config.EnforceResourceVerificationMode,
+			expected:                 signedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "tampered task with enforce policy",
+			requester:                requesterTampered,
+			resourceVerificationMode: config.EnforceResourceVerificationMode,
+			expected:                 nil,
+			expectedErr:              trustedresources.ErrorSignatureMismatch,
+		}, {
+			name:                     "unsigned task with warn policy",
+			requester:                requesterUnsigned,
+			resourceVerificationMode: config.WarnResourceVerificationMode,
+			expected:                 unsignedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "signed task with warn policy",
+			requester:                requesterSigned,
+			resourceVerificationMode: config.WarnResourceVerificationMode,
+			expected:                 signedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "tampered task with warn policy",
+			requester:                requesterTampered,
+			resourceVerificationMode: config.SkipResourceVerificationMode,
+			expected:                 tamperedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "unsigned task with skip policy",
+			requester:                requesterUnsigned,
+			resourceVerificationMode: config.SkipResourceVerificationMode,
+			expected:                 unsignedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "signed task with skip policy",
+			requester:                requesterSigned,
+			resourceVerificationMode: config.SkipResourceVerificationMode,
+			expected:                 signedTask,
+			expectedErr:              nil,
+		}, {
+			name:                     "tampered task with skip policy",
+			requester:                requesterTampered,
+			resourceVerificationMode: config.WarnResourceVerificationMode,
+			expected:                 tamperedTask,
+			expectedErr:              nil,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx = ttesting.SetupTrustedResourceConfig(ctx, secretpath, tc.resourceVerificationMode)
+			tr := &v1beta1.TaskRun{
+				ObjectMeta: metav1.ObjectMeta{Namespace: "trusted-resources"},
+				Spec: v1beta1.TaskRunSpec{
+					TaskRef:            taskRef,
+					ServiceAccountName: "default",
+				},
+			}
+			fn, err := resources.GetTaskFunc(ctx, nil, nil, tc.requester, tr, tr.Spec.TaskRef, "", "default", "default")
+			if err != nil {
+				t.Fatalf("failed to get task fn: %s", err.Error())
+			}
+
+			resolvedTask, err := fn(ctx, taskRef.Name)
+
+			if tc.expectedErr != nil && (err == nil || err.Error() != tc.expectedErr.Error()) {
+				t.Fatalf("Expected error %v but found %v instead", tc.expectedErr, err)
+			} else if tc.expectedErr == nil && err != nil {
+				t.Fatalf("Received unexpected error ( %#v )", err)
+			}
```

Comment:
nit: Since you're always comparing against a known error type, you could simplify this by comparing the error directly / using `errors.Is` - https://go.dev/play/p/PpZrSvYxbDK

Reply:
Thanks!